.github/workflows/step_tests-pip.yml

```diff
@@ -1,54 +1,69 @@
 name: test-pip
-run-name: Run tests using Pip
+run-name: Run main tests using Pip
 
 on:
   workflow_call:
+    inputs:
+      coverage:
+        type: boolean
+        description: Upload coverage to CodeCov
+        default: true
 
-concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}-pip-test
-  cancel-in-progress: true
+permissions:
+  contents: read
 
 jobs:
   test-pip:
+    name: >
+      🐍 ${{ matrix.python-version }}
+      ${{ matrix.dependency_type && format('({0})', matrix.dependency_type) }}
+      ${{ matrix.markdown-it-py && format('(markdown-it-py {0})', matrix.markdown-it-py) }}
+      ${{ matrix.quarto && '(Quarto)' }}
+      ${{ matrix.no_kernel && '(No kernel)' }}
     runs-on: ubuntu-latest
 
     strategy:
+      fail-fast: false
       matrix:
         python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12"]
-        markdown-it-py-version: ["~=3.0"]
-        pip-flags: [""]
         include:
+          # Test pre-release versions
           - python-version: "3.x"
-            pip-flags: "--pre --upgrade --upgrade-strategy eager"
+            dependency_type: "pre"
```

Comment:
Thanks for this! At the moment the `pre`-pip tests are not passing (an issue with the `pytest` release candidate). Do you think there is a way to mark this step as experimental, i.e. get a red box for that particular one but still get a pass globally?

Reply:
Yes, I've recently designed a workflow like that, I'll add it here too: https://github.com/LecrisUT/CMake-Template/blob/a4f73dbd53d8cb98e6325296c2905b2ed40dcaa3/.github/workflows/step_test.yaml#L46

Reply:
Interesting! I am not familiar with the inputs yet but yes this seems to be the way to go. I've added the experimental flag in this PR, on top of yours: https://github.com/mwouts/jupytext/pull/1204